src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,94 @@
+        if (names.length > 0) {
+            int[] incomes = new int[names.length];
+            for (String dataElement : data) {
+                String[] temporarilyDividedData = dataElement.split(DATA_ELEMENTS_SPLITERATOR);
+                LocalDate dateFromData =
+                        parseDateFromData(temporarilyDividedData[DATE_POSITION]);
+                String nameFromDataPosition = temporarilyDividedData[NAME_POSITION];
+                int hoursOfWork = parseIntegerFromData(temporarilyDividedData,
+                        HOURS_OF_WORK_POSITION);
+                int incomePerHourPosition = parseIntegerFromData(temporarilyDividedData,
+                        INCOME_PER_HOUR_POSITION);
+                int incomeFromData = hoursOfWork * incomePerHourPosition;
+                if (isDateSufficient(dateFromData, localDateFrom, localDateTo)) {
+                    int index = indexOfValue(names, nameFromDataPosition);
+                    if (index >= 0) {
+                        incomes[index] += incomeFromData;
+                    }
+                }
+            }
+            return generateReport(names, incomes, dateFrom, dateTo);
+        } else {
+            throw new EmptyNamesListException("Names list length should have been at least 1"
+                    + " to generate a report but was " + names.length);
+        }
+    }
+
+    private static LocalDate parseDateFromData(String date) {
+        try {
+            return LocalDate.parse(date, FORMATTER);
+        } catch (DateTimeParseException dtpe) {
+            throw new RuntimeException("Couldn't parse parameter: " + date);
+        }
+    }
+
+    private static int parseIntegerFromData(String[] temporarilyDividedData, int position) {
+        try {
+            return Integer.parseInt(temporarilyDividedData[position]);
+        } catch (NumberFormatException nfe) {
+            throw new RuntimeException(
+                    "Couldn't parse integer for " + temporarilyDividedData[position]);
+        }
+    }
+
+    private static boolean isDateSufficient(LocalDate dateFromData, LocalDate localDateFrom,
+                                            LocalDate localDateTo) {
+        return (dateFromData.isAfter(localDateFrom)
+                || dateFromData.isEqual(localDateFrom))
+                && (dateFromData.isBefore(localDateTo)
+                || dateFromData.isEqual(localDateTo));
+    }
+
+    private static int indexOfValue(String[] names, String targetName) {
+        for (int i = 0; i < names.length; i++) {
+            if (names[i].equals(targetName)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private static String generateReport(String[] names, int[] incomes, String dateFrom,
+                                         String dateTo) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo)
```

Comment:
do not use such naming(`builder`), how to guess what it stores?
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
        StringBuilder report = new StringBuilder();
        builder.append(HEADER).append(dateFrom).append(SEPARATOR).append(dateTo)
```